sbol_utilities/component.py

```diff
@@ -246,6 +246,8 @@ def rna_component_with_sequence(identity: str, sequence: str, **kwargs)-> Tuple[
     rna_comp = sbol3.Component(identity, sbol3.SBO_RNA,
                                 sequences =[comp_seq],
                                 **kwargs)
+    if rna_comp.name == None:
+        rna_comp.name = identity
```

Comment:
I don't think that you want to put a URI in as the names. "https://www.ncbi.nlm.nih.gov/nuccore/ATP07149_1" is not a very good name. I think you're looking for the `display_id`. I think it would be better, though, to just leave the name blank, as the spec says:

> If an Identified object lacks a name, then software tools SHOULD instead display the object’s displayId or URI.

If you set the name, then you are saying "this is the right name", if you leave it unset, then you are saying "the user didn't give a name, so this object should fail over to showing display_id"

Reply:
> I don't think that you want to put a URI in as the names. "https://www.ncbi.nlm.nih.gov/nuccore/ATP07149_1" is not a very good name.

In ExternallyDefined objects the URI is the definition, the name can be setted using **kwargs if I understood the first part. The second argument is very good  so I rolled back the if statements that copies identity  on name.